statistics/handle/lock_stats_handler.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pingcap/tidb/statistics/handle/cache"
+	"github.com/pingcap/tidb/util/logutil"
+	"github.com/pingcap/tidb/util/sqlexec"
+	"go.uber.org/zap"
+)
+
+// AddLockedTables add locked tables id to store
+func (h *Handle) AddLockedTables(tids []int64, pids []int64, tables []*ast.TableName) (string, error) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnStats)
+	exec := h.mu.ctx.(sqlexec.SQLExecutor)
+
+	_, err := exec.ExecuteInternal(ctx, "BEGIN PESSIMISTIC")
+	if err != nil {
+		return "", err
+	}
+
+	// Load tables to check duplicate when inserting.
+	rows, _, err := h.execRestrictedSQL(ctx, "SELECT table_id FROM mysql.stats_table_locked")
```

Comment:
It's a little weird here... First, we use `exec.Execute` to begin a new transaction, and then use another API `h.exec` to run a query. I don't know whether this is on purpose.

Reply:
If we change `h.exec` to `exec.Execute`, will it cause any failure? I just looked at some other functions like `SaveTableStatsToStorage`, it seems like we should always use `exec.Execute` here.

Reply:
Updated in [`ccb67d3` (#46403)](https://github.com/pingcap/tidb/pull/46403/commits/ccb67d3edde7aafd84996a8659ad677a479e3c64)